test/hotspot/jtreg/compiler/lib/verify/Verify.java

```diff
@@ -203,82 +302,239 @@ private static void checkEQimpl(MemorySegment a, MemorySegment b, String context
+        }
+
+        for (int i = 0; i < a.length; i++) {
+            if (!isDoubleEQ(a[i], b[i])) {
+                System.err.println("ERROR: Verify.checkEQ failed: value mismatch at " + i + ": " + a[i] + " vs " + b[i] + ". check raw: " + isFloatCheckWithRawBits);
+                print(a, b, field, aParent, bParent);
+                throw new VerifyException("Double array value mismatch " + a[i] + " vs " + b[i]);
+            }
+        }
     }
 
     /**
-     * Verify that the content of two double arrays is identical.
+     * Verify that the contents of two boolean arrays are identical.
      */
-    private static void checkEQimpl(double[] a, double[] b, String context) {
-        checkEQimpl(MemorySegment.ofArray(a), MemorySegment.ofArray(b), context);
+    private void checkEQimpl(boolean[] a, boolean[] b, String field, Object aParent, Object bParent) {
+        if (a.length != b.length) {
+            System.err.println("ERROR: Verify.checkEQ failed: length mismatch: " + a.length + " vs " + b.length);
+            print(a, b, field, aParent, bParent);
+            throw new VerifyException("Boolean array length mismatch.");
+        }
+
+        for (int i = 0; i < a.length; i++) {
+            if (a[i] != b[i]) {
+                System.err.println("ERROR: Verify.checkEQ failed: value mismatch at " + i + ": " + a[i] + " vs " + b[i]);
+                print(a, b, field, aParent, bParent);
+                throw new VerifyException("Boolean array value mismatch.");
+            }
+        }
     }
 
     /**
-     * Verify that the content of two Object arrays is identical, recursively:
+     * Verify that the contents of two Object arrays are identical, recursively:
      * every element is compared with checkEQimpl for the corresponding type.
      */
-    private static void checkEQimpl(Object[] a, Object[] b, String context) {
+    private void checkEQimpl(Object[] a, Object[] b, String field, Object aParent, Object bParent) {
         // (1) Length mismatch
         if (a.length != b.length) {
             System.err.println("ERROR: Verify.checkEQ failed: length mismatch: " + a.length + " vs " + b.length);
+            print(a, b, field, aParent, bParent);
             throw new VerifyException("Object array length mismatch.");
         }
 
         for (int i = 0; i < a.length; i++) {
             // Recursive checkEQ call.
-            checkEQ(a[i], b[i], "[" + i + "]" + context);
+            checkEQdispatch(a[i], b[i], "[" + i + "]", a, b);
         }
     }
 
-    private static void print(Object a, String context) {
-        if (a == null) {
-            System.err.println("  " + context + ": null");
+    /**
+     * We do not want to import jdk.incubator.vector explicitly, because it would mean we would also have
+     * to add "--add-modules=jdk.incubator.vector" to the command-line of every test that uses the Verify
+     * class. So we hack this via reflection.
+     */
```

Comment:
I think this background is only needed at `checkEQForVectorAPIClass()` (where you already have that comment). Here you can just describe what the code actually does or just drop the comment entirely since the method name is self-explanatory :-)